### PR TITLE
Update UG to include new commands and updated commands in v1.2

### PR DESCRIPTION
**Pull request by appleraincoat**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -53,34 +53,58 @@ command to run the application.<br>
 
 ### Adding a client: `add`
 
-Adds a client to the Realodex. Note that Name, Address, Income, Phone Number, Family Size, Email and Tag are compulsory. You may input the parameters in any order (e.g. if 
-the command specifies `n/NAME a/ADDRESS`, `a/ADDRESS n/NAME` is also acceptable). A client can be a buyer, seller or both. This should be reflected by tagging `t/buyer` and/or `t/seller`.
+Adds a client to the Realodex. 
 
-Format: `add n/NAME p/PHONE i/INCOME e/EMAIL a/ADDRESS f/FAMILY t/TAG ...`
+Format: `add n/NAME p/PHONE i/INCOME e/EMAIL a/ADDRESS f/FAMILY t/TAG [r/REMARK]`
```
Comment (wapisai) [inline docs/UserGuide.md:58]: Did you put the [] to show that remark is optional?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -53,34 +53,58 @@ command to run the application.<br>
...
+- You may input the parameters in any order (e.g. if
+the command specifies `n/NAME a/ADDRESS`, `a/ADDRESS n/NAME` is also acceptable). 
+- Note that the tag is to indicate if a client is a
+Buyer, Seller or both, so tags only accept "buyer" or "seller" as the input (case-insensitive).
 
 Examples:
-* `add n/John Doe p/98765432 i/20000 e/johnd@example.com a/311, Clementi Ave 2, #02-25 f/4 t/buyer`
-* `add n/Betsy Crowe a/Newgate Prison i/$0 f/1 p/94859694 e/betsyc@rocketmail.com t/seller`
+* `add n/John Doe p/98765432 i/20000 e/johnd@example.com a/311, Clementi Ave 2, #02-25 f/4 t/Buyer r/Owes $1000.`
+* `add n/Betsy Crowe a/Newgate Prison i/$0 f/1 p/94859694 e/betsyc@rocketmail.com t/Seller`
```
Comment (wapisai) [inline docs/UserGuide.md:68]:
```suggestion
* `add n/Betsy Crowe a/Newgate Prison i/$0 f/1 p/94859694 e/betsyc@rocketmail.com t/Seller t/Buyer`
```

Maybe could use this example to show how clients can be both buyers and sellers

File: docs/UserGuide.md [hunk 3]
```diff
@@ -53,34 +53,58 @@ command to run the application.<br>
...
 ### Deleting a client : `delete`
 
-Deletes the specified client from the Realodex.
+Deletes the specified client from the Realodex. There are 2 ways to do so: 
 
-Format: `delete NAME`
+#### Deleting by name
 
-Enter client's `NAME` after prompt is shown.
+Format: `delete NAME`
```
Comment (wapisai) [inline docs/UserGuide.md:76]:
```suggestion
Format: `delete n/NAME`
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -53,34 +53,58 @@ command to run the application.<br>
...
 
 * Deletes the client of the specified `NAME`.
 * If name is **not found**, error message will be shown `"NAME" is not found`.
 
-Examples:
-* `delete` followed by `Udhaya Shanmugam` deletes the client in the Realodex with the name "Udhaya Shanmugam".
+Example:
+* `delete Udhaya Shanmugam` deletes the client in the Realodex with the name "Udhaya Shanmugam".
```
Comment (wapisai) [inline docs/UserGuide.md:82]:
```suggestion
* `delete n/Udhaya Shanmugam` deletes the client in the Realodex with the name "Udhaya Shanmugam".
```

File: docs/UserGuide.md [hunk 5]
```diff
@@ -92,6 +116,22 @@ Format: `filter KEYPHRASE`
 - Partial words will still be matched e.g. `Udh` will match `Udhaya`
 - All persons' names containing the keyword will be returned e.g. `Al` will return `Alicia`, `Allysa`
 
+Example:
+
+- `filter Al` will list out persons whose name has `"Al"` inside, such as `"Alicia"`, `"Allysa"` and `"Jamal""`
```
Comment (wapisai) [inline docs/UserGuide.md:121]: the bros

File: docs/UserGuide.md [hunk 6]
```diff
@@ -137,7 +177,11 @@ _Details coming soon ..._
 
 Action     | Format, Examples
 -----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------
-**Add**    | `add n/NAME p/PHONE i/INCOME e/EMAIL a/ADDRESS f/FAMILY [t/TAG]...` <br> e.g., `add n/John Doe p/98765432 i/20000 e/johnd@example.com a/311, Clementi Ave 2, #02-25 f/4 t/friends t/owesMoney`
-**Delete** | `delete NAME`<br> e.g., `delete John`
+**Add**    | `add n/NAME p/PHONE i/INCOME e/EMAIL a/ADDRESS f/FAMILY t/TAG [r/REMARK]` <br> e.g. `add n/John Doe p/98765432 i/20000 e/johnd@example.com a/311, Clementi Ave 2, #02-25 f/4 r/Buyer t/Owes $1000.`
+**Delete (by name)** | `delete NAME`<br> e.g. `delete John`
```
Comment (wapisai) [inline docs/UserGuide.md:181]:
```suggestion
**Delete (by name)** | `delete n/NAME`<br> e.g. `delete n/John`
```

**Review by wapisai — COMMENTED** [review]
Just small problems! Otherwise LGTM!

File: docs/UserGuide.md [hunk 7]
```diff
@@ -30,10 +30,10 @@ command to run the application.<br>
 
 1. Some example commands you can try:
 
-   * `add n/John Doe p/98765432 i/20000 e/johnd@example.com a/311, Clementi Ave 2, #02-25 f/4 t/friends t/owesMoney`
+   * `add n/John Doe p/98765432 i/20000 e/johnd@example.com a/311, Clementi Ave 2, #02-25 f/4 t/buyer r/Owes money.`
    Adds a contact named `John Doe` to the Realodex.
```
Comment (UdhayaShan1) [inline docs/UserGuide.md:34]: Thanks

File: docs/UserGuide.md [hunk 8]
```diff
@@ -30,10 +30,10 @@ command to run the application.<br>
...
 
-   * `delete John Doe` : Deletes the client with name `John Doe` from the current list.
+   * `delete John Doe` : Deletes the client with name `John Doe` from the Realodex.
```
Comment (UdhayaShan1) [inline docs/UserGuide.md:36]:
```suggestion
   * `delete John Doe` : Deletes the client with name `John Doe` from Realodex.
```

File: docs/UserGuide.md [hunk 9]
```diff
@@ -53,34 +53,58 @@ command to run the application.<br>
 
 ### Adding a client: `add`
 
-Adds a client to the Realodex. Note that Name, Address, Income, Phone Number, Family Size, Email and Tag are compulsory. You may input the parameters in any order (e.g. if 
-the command specifies `n/NAME a/ADDRESS`, `a/ADDRESS n/NAME` is also acceptable). A client can be a buyer, seller or both. This should be reflected by tagging `t/buyer` and/or `t/seller`.
+Adds a client to the Realodex.
```
Comment (UdhayaShan1) [inline docs/UserGuide.md:56]:
```suggestion
Adds a client to Realodex. 
```

File: docs/UserGuide.md (hunk 1 above)
Comment (UdhayaShan1) [reply docs/UserGuide.md:58]: Yes

Comment (UdhayaShan1) [reply docs/UserGuide.md:58]: Can highlight to show [] is optional thanks

File: docs/UserGuide.md (hunk 5 above)
Comment (UdhayaShan1) [reply docs/UserGuide.md:121]: Jamal

**Review by UdhayaShan1 — CHANGES_REQUESTED** [review]
Again @wapisai comments, thanks yaposoks

File: docs/UserGuide.md [hunk 10]
```diff
@@ -53,34 +53,58 @@ command to run the application.<br>
...
 Examples:
-* `add n/John Doe p/98765432 i/20000 e/johnd@example.com a/311, Clementi Ave 2, #02-25 f/4 t/buyer`
-* `add n/Betsy Crowe a/Newgate Prison i/$0 f/1 p/94859694 e/betsyc@rocketmail.com t/seller`
+* `add n/John Doe p/98765432 i/20000 e/johnd@example.com a/311, Clementi Ave 2, #02-25 f/4 t/Buyer r/Owes $1000.`
+* `add n/Betsy Crowe a/Newgate Prison i/$0 f/1 p/94859694 e/betsyc@rocketmail.com t/Seller`
 
 ### Deleting a client : `delete`
 
-Deletes the specified client from the Realodex.
+Deletes the specified client from the Realodex. There are 2 ways to do so:
```
Comment (4llysa) [inline docs/UserGuide.md:72]: There is only one way to do so. Delete by name is not implemented (from our previous discussion, i recall that delete by name is not implemented for consistency with edit that only allows edit by index)

File: docs/UserGuide.md [hunk 11]
```diff
@@ -53,34 +53,58 @@ command to run the application.<br>
...
 
 ### Deleting a client : `delete`
 
-Deletes the specified client from the Realodex.
+Deletes the specified client from the Realodex. There are 2 ways to do so: 
 
-Format: `delete NAME`
+#### Deleting by name
 
-Enter client's `NAME` after prompt is shown.
```
Comment (4llysa) [inline docs/UserGuide.md:71]: can remove delete by name feature unless there is strong inclination towards implementing delete by name for v1.2

**Review by UdhayaShan1 — APPROVED** [review]
LGTM! Thanks for editing!

File: docs/UserGuide.md (hunk 5 above)
Comment (jyztintan) [reply docs/UserGuide.md]: +1

**Pull request merged**